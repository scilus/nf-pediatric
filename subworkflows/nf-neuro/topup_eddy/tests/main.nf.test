nextflow_workflow {

    name "Test Subworkflow TOPUP_EDDY"
    script "../main.nf"
    workflow "TOPUP_EDDY"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/topup_eddy"

    tag "preproc"
    tag "preproc/topup"
    tag "preproc/eddy"
    tag "utils/extractb0"
    tag "betcrop/fslbetcrop"

    tag "load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "topup_eddy_light.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("topup_eddy - ONLY Eddy - AP DWI | PA sbref") {

        when {
            workflow {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.nii.gz"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bval"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bvec")
                    ]}
                input[1] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-AP_sbref.nii.gz")
                    ]}
                input[2] = Channel.from( [] )
                input[3] = Channel.from( [] )
                input[4] = Channel.from( [] )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out
                        .findAll{ !it.key.isInteger() && it.value  }
                        .collect{ out -> ["b0_mask", "bval", "versions"].contains(out.key)
                        ? out.value
                        : file(out.value.get(0).get(1)).name }
                ).match() }
            )
        }
    }

    test("topup_eddy - AP DWI | PA sbref") {
        when {
            workflow {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.nii.gz"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bval"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bvec")
                    ]}
                input[1] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-AP_sbref.nii.gz")
                    ]}
                input[2] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        [],
                        [],
                        []
                    ]}

                input[3] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-PA_sbref.nii.gz")
                    ]}
                input[4] = Channel.from( [] )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out
                        .findAll{ !it.key.isInteger() && it.value  }
                        .collect{ out -> ["b0_mask", "bval", "versions"].contains(out.key)
                        ? out.value
                        : file(out.value.get(0).get(1)).name }
                ).match() }
            )
        }
    }

    test("topup_eddy - AP DWI+sbref | PA DWI+sbref") {

        when {
            workflow {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.nii.gz"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bval"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bvec")
                    ]}
                input[1] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-AP_sbref.nii.gz")
                    ]}
                input[2] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-PA_dwi.nii.gz"),
                        file("\${test_data_directory}/sub-01_dir-PA_dwi.bval"),
                        file("\${test_data_directory}/sub-01_dir-PA_dwi.bvec")
                    ]}
                input[3] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-PA_sbref.nii.gz")
                    ]}
                input[4] = Channel.from( [] )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out
                        .findAll{ !it.key.isInteger() && it.value  }
                        .collect{ out -> ["b0_mask", "bval", "versions"].contains(out.key)
                        ? out.value
                        : file(out.value.get(0).get(1)).name }
                ).match() }
            )
        }
    }
}
