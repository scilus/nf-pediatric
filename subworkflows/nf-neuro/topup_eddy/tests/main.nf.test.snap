{
    "topup_eddy - AP DWI | PA sbref": {
        "content": [
            [
                "test_b0.nii.gz",
                [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__b0_bet_mask.nii.gz:md5,1b45aae94d5bb093dca6ca4a71a8bf97"
                    ]
                ],
                [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_eddy_corrected.bval:md5,8192e97b08e8032382397bb844b31892"
                    ]
                ],
                "test__dwi_eddy_corrected.bvec",
                "test__dwi_corrected.nii.gz",
                [
                    "versions.yml:md5,4b74184cf640fe471fd336a12cd7d565",
                    "versions.yml:md5,dc9478b4d3cb93d93799e493aefe5220"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-01T00:20:35.227281757"
    },
    "topup_eddy - ONLY Eddy - AP DWI | PA sbref": {
        "content": [
            [
                "test_b0.nii.gz",
                [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__b0_bet_mask.nii.gz:md5,1b45aae94d5bb093dca6ca4a71a8bf97"
                    ]
                ],
                [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_eddy_corrected.bval:md5,8192e97b08e8032382397bb844b31892"
                    ]
                ],
                "test__dwi_eddy_corrected.bvec",
                "test__dwi_corrected.nii.gz",
                [
                    "versions.yml:md5,4b74184cf640fe471fd336a12cd7d565",
                    "versions.yml:md5,dc9478b4d3cb93d93799e493aefe5220"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-01T00:19:48.37302071"
    },
    "topup_eddy - AP DWI+sbref | PA DWI+sbref": {
        "content": [
            [
                "test_b0.nii.gz",
                [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__b0_bet_mask.nii.gz:md5,1b45aae94d5bb093dca6ca4a71a8bf97"
                    ]
                ],
                [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_eddy_corrected.bval:md5,8192e97b08e8032382397bb844b31892"
                    ]
                ],
                "test__dwi_eddy_corrected.bvec",
                "test__dwi_corrected.nii.gz",
                [
                    "versions.yml:md5,4b74184cf640fe471fd336a12cd7d565",
                    "versions.yml:md5,dc9478b4d3cb93d93799e493aefe5220"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-01T00:21:19.437875741"
    }
}